src/main/java/com/hillel/items_exchange/controller/SubcategoryController.java

```diff
@@ -13,32 +12,28 @@
 
 import static com.hillel.items_exchange.config.SecurityConfig.HAS_ROLE_ADMIN;
 import static com.hillel.items_exchange.util.MessageSourceUtil.getExceptionMessageSourceWithId;
+import io.swagger.annotations.ApiOperation;
 
 @RestController
 @RequestMapping("/subcategory")
-@CrossOrigin(origins = "http://localhost:4200")
 @RequiredArgsConstructor
-@Slf4j
 public class SubcategoryController {
 
     private final SubcategoryService subcategoryService;
 
     @GetMapping("/{category_id}/names")
-    public ResponseEntity<List<String>> getSubcategoryNamesByCategoryId(@PathVariable("category_id") long id) {
-        List<String> subcategoriesNames = subcategoryService.findSubcategoryNamesByCategoryId(id);
-        if (subcategoriesNames.isEmpty()) {
-            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-        }
-
-        return new ResponseEntity<>(subcategoriesNames, HttpStatus.OK);
+    @ResponseStatus(HttpStatus.OK)
```

Comment:
Because Category and Subcategory entities are simple and always connected, and to simplify the business logic we decided to create/update subcategory inside the Category methods.